src/ripple/app/main/LoadManager.cpp

```diff
@@ -22,192 +22,180 @@
 #include <ripple/app/main/Application.h>
 #include <ripple/app/misc/NetworkOPs.h>
 #include <ripple/basics/UptimeTimer.h>
-#include <ripple/core/JobQueue.h>
 #include <ripple/core/LoadFeeTrack.h>
 #include <ripple/json/to_string.h>
 #include <beast/threads/Thread.h>
-#include <beast/cxx14/memory.h> // <memory>
-#include <mutex>
-#include <thread>
 
 namespace ripple {
 
-class LoadManagerImp
-    : public LoadManager
-    , public beast::Thread
+LoadManager::LoadManager (
+    Application& app, Stoppable& parent, beast::Journal journal)
+    : Stoppable ("LoadManager", parent)
+    , app_ (app)
+    , journal_ (journal)
+    , deadLock_ (0)
+    , armed_ (false)
+    , stop_ (false)
 {
-public:
-    using LockType = std::mutex;
-    using ScopedLockType = std::lock_guard <LockType>;
-
-    Application& app_;
-    beast::Journal m_journal;
-    LockType mLock;
-
-    bool mArmed;
-
-    int mDeadLock;              // Detect server deadlocks
-    //--------------------------------------------------------------------------
-
-    LoadManagerImp (Application& app,
-            Stoppable& parent, beast::Journal journal)
-        : LoadManager (parent)
-        , Thread ("loadmgr")
-        , app_ (app)
-        , m_journal (journal)
-        , mArmed (false)
-        , mDeadLock (0)
-    {
-        UptimeTimer::getInstance ().beginManualUpdates ();
-    }
+    UptimeTimer::getInstance ().beginManualUpdates ();
+}
 
-    ~LoadManagerImp ()
+LoadManager::~LoadManager ()
+{
+    try
     {
         UptimeTimer::getInstance ().endManualUpdates ();
-
-        stopThread ();
+        onStop ();
     }
-
-    //--------------------------------------------------------------------------
-    //
-    // Stoppable
-    //
-    //--------------------------------------------------------------------------
-
-    void onPrepare ()
+    catch (std::exception const& ex)
     {
+        // Swallow the exception in a destructor.
+        JLOG(journal_.warning) << "std::exception in ~LoadManager.  "
+            << ex.what();
     }
-
-    void onStart ()
+    catch (...)
```

Comment:
We need to move away from `catch(...)` statements, because they goof up `Boost.Coroutine` (they swallow the `boost::coroutine::forced_unwind` exception instead of propagating it).